plugins/node/opentelemetry-instrumentation-aws-lambda/src/aws-lambda.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as path from 'path';
+
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+  safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+import {
+  diag,
+  Span,
+  SpanKind,
+  SpanStatusCode,
+  TracerProvider,
+} from '@opentelemetry/api';
+import { CLOUD_RESOURCE } from '@opentelemetry/resources';
+import { FaasAttribute } from '@opentelemetry/semantic-conventions';
+
+import { Callback, Context, Handler } from 'aws-lambda';
+
+import { LambdaModule } from './types';
+import { VERSION } from './version';
+import { NodeTracerProvider } from '@opentelemetry/node';
+
+export class AwsLambdaInstrumentation extends InstrumentationBase {
+  private _tracerProvider: TracerProvider | undefined;
+
+  constructor() {
+    super('@opentelemetry/instrumentation-awslambda', VERSION);
+  }
+
+  init() {
+    // _HANDLER and LAMBDA_TASK_ROOT are always defined in Lambda.
+    const taskRoot = process.env.LAMBDA_TASK_ROOT!;
```

Comment:
Maybe skip init if this is not true? e.g. if AWS changes this or someone loads this extension somewhere outside of AWS?